packages/hash/api/src/collab/save.ts

```diff
@@ -126,6 +126,57 @@ const calculateSaveActions = async (
   const draftIdToPlaceholderId = new Map<string, string>();
   const draftIdToBlockEntities = new Map<string, DraftEntity<BlockEntity>>();
 
+  const shouldSetTextProperties = <T extends {}>(
+    properties: T,
+    savedProperties?: DraftEntity["properties"],
+  ): (T & { text: Omit<LegacyLink<DraftEntity<Text>>, "data"> }) | null => {
+    if (!isDraftTextContainingEntityProperties(properties)) {
+      return null;
+    }
+
+    const savedWithoutText = omit(savedProperties, "text");
+    const savedTextLink = isDraftTextContainingEntityProperties(savedProperties)
+      ? savedProperties.text.__linkedData
+      : null;
+
+    if (savedTextLink) {
+      const draftWithoutText = omit(properties, "text");
+
+      if (
+        !isEqual(savedWithoutText, draftWithoutText) ||
+        isEqual(savedTextLink, properties.text.__linkedData)
+      ) {
```

Comment:
I think this might be a bug, but I'm not actually certain. It's a replication of the same logic in main. The idea here is we don't want to update an entity just because the text entity within it has changed (as that is processed separately), but I think I got it wrong originally and therefore in this refactor.

I've rewritten this whole process in [`b807520` (#872)](https://github.com/hashintel/hash/pull/872/commits/b807520cb738d9b1d0c6b25adcb63fb5d50ceb2f) to make this significantly clearer

Reply:
Looks good, thank you